### PR TITLE
Pre-release fixes and syncing with latest DSL2 pipeline template

### PR DESCRIPTION
**Pull request by drpatelh**

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -275,27 +274,100 @@ You can also supply a run name to resume a specific run: `-resume [run-name]`. U
...
+Command executed:
+    STAR \
+        --genomeDir star \
+        --readFilesIn WT_REP1_trimmed.fq.gz  \
+        --runThreadN 2 \
+        --outFileNamePrefix WT_REP1. \
+        <TRUNCATED>
+
+Command exit status:
+    137
```
Comment (aunderwo) [inline docs/usage.md:302]: @drpatelh is the use of the rnaseq pipeline here just an example?

File: docs/usage.md [hunk 2]
```diff
@@ -275,27 +274,100 @@ You can also supply a run name to resume a specific run: `-resume [run-name]`. U
...
+
+```nextflow
+params {
+    modules {
+        'star_align' {
+            args          = "--quantMode TranscriptomeSAM --twopassMode Basic --outSAMtype BAM Unsorted --readFilesCommand zcat --runRNGseed 0 --outFilterMultimapNmax 20 --alignSJDBoverhangMin 1 --outSAMattributes NH HI AS NM MD --quantTranscriptomeBan Singleend --outFilterMismatchNmax 16"
+            publish_dir   = "my_star_directory"
+            publish_files = ['out':'log', 'tab':'log', 'bam':'']
+        }
+    }
```
Comment (aunderwo) [inline docs/usage.md:358]: There are a lot of lines here about staralign - I worry a little they detract from the main usage. Could/Should these be made more specific or moved to a different md file?

**Review by aunderwo — COMMENTED** [review]
@drpatelh 

Thanks for all your hard work putting in effort to bringing this up to latest DSL2 standards/conventions. I have just a couple of comments about the `usage.md`

File: docs/usage.md (hunk 1 above)
Comment (drpatelh) [reply docs/usage.md:302]: Yep, exactly. We needed a real-world example to highlight how to tweak these resources.

File: docs/usage.md (hunk 2 above)
Comment (drpatelh) [reply docs/usage.md:358]: So the idea behind adding these particular docs and using `STAR_ALIGN` as an example is that it forms a general narrative to users as to how to change the resources for DSL2 pipelines and to amend the module options. I needed to find a process that we could use throughout to be coherent with the explanations and to be able to point to real code in the rnaseq pipeline that would help users troubleshoot and figure this sort of thing out for themselves - `STAR_ALIGN` is one of those processes that has to be tweaked quite often which is why I chose that as an example. Took me the best part of the day to write these two paragraphs! But they have been pointed to numerous times already where users have wanted to tweak such options. I digress, you are absolutely right about moving the docs - we have discussed this a few times but we don't have an official home for them yet, at which point they will be moved out of the DSL2 pipeline template 👍🏽

**Review by aunderwo — APPROVED** [review]

**Pull request merged**